packages/common/src/transformers/VizConfigTransformers/Table/TableConfigTransformer.ts

```diff
@@ -55,19 +84,19 @@ export class TableConfigTransformer<
      * @param currentColumns
      * @private
      */
-    private getColumns(
-        currentColumns?: TableConfig['columns'],
-    ): TableConfig['columns'] {
+    private getColumns(currentColumns?: TableConfig['columns']) {
         return this.resultsTransformer
             .getFieldOptions()
-            .reduce<TableConfig['columns']>((acc, field) => {
-                acc[field] = {
-                    visible: true,
-                    name: field,
-                    frozen: false,
-                    ...(currentColumns ?? {})[field],
-                };
-                return acc;
-            }, {});
+            .map<TableConfig['columns'][number]>(
+                (fieldId) =>
+                    currentColumns?.find(
+                        (column) => column.fieldId === fieldId,
+                    ) ?? {
+                        fieldId,
+                        visible: true,
+                        frozen: false,
+                        name: fieldId,
+                    },
+            );
```

Comment:
I changed the type because "tool calling" struggles with json schema with "additionalProperties".

[fieldId]: object --became--> array of {...object, fieldId}